### ISSUE TITLE
Error while turning on Settings Sync. No authentication providers are available.

### ISSUE THREAD

**Issue by tlockney** [microsoft/vscode-remote-release#7001]
- VSCode Version: server (vscode-server-launcher/20220727.1) / UI 1.70.0-insider (though this is occurring on both insider and stable versions of the UI)
- Local OS Version: OSX 12.4
- Remote OS Version: OSX 12.4
- Remote Extension/Connection Type: Server
- Logs: 

Steps to Reproduce:

1. Connect to remote via https://insiders.vscode.dev/+ms-vscode.remote-server/[myservername]:31545
2. See that settings aren't being synced and there's no prompt to turn it on3. 
3. Open command palette and select 'Settings Sync: Show Settings' which then shows the button to 'Turn on Settings Sync'. 
4. Click button and see error message: 'Error while turning on Settings Sync. No authentication providers are available.'

Does this issue occur when you try this locally?: No, this only seems to happen when using the remote server (that is, starting with `code server serve` method)
Does this issue occur when you try this locally and all extensions are disabled?: No

**Comment by bamurtaugh** [comment]
Thanks for reporting @tlockney! If you're still experiencing this, do you potentially see any updates (run `code-server update`)?

Wondering if this is macOS-specific, as I'm not seeing it in WSL.

**Comment by tlockney** [comment]
I ran `code-server update` but it didn't seem to find any updates available and still shows the version I listed above. Is there supposed to be a newer update available? I'm still seeing the same behavior (currently running `code-server` on an m1 Mac mini and accessing it from my M2 MacBook Air).

**Comment by bamurtaugh** [comment]
Thanks for checking, I believe that's the latest (@tanhakabir can correct me if not 😄).

@sandy081 @TylerLeonhardt any thoughts?

**Comment by TylerLeonhardt** [comment]
Can you follow these steps so that we can get more information on what's going on?

* Open the Output pane (ctrl+shift+u I think on Windows)
* drop in this issue he contents of the following:
  * `GitHub Authentication` or `Microsoft Authentication` Output channel
  * `Log (Settings Sync)` Output channel
  * `Log (Window)` Output channel

**Comment by tlockney** [comment]
I just tried this and couldn't find any sign of `GitHub Authentication` or `Microsoft Authentication`. So then I tried it in private browsing session, but got the same results. But then I tried it in Chrome (up until now I've been testing in Safari since it has solid PWA support on OSX). In Chrome everything seems to work (though I do see some 404s in the dev tools console, but those show up in both Safari and Chrome). 

So it sounds like this might be limited to just Safari (I don't currently have Firefox on this machine to test with). I doubt it will tell you anything useful, but from Safari, here's the output for the `Log (Settings Sync)` channel:

```
[2022-08-08 15:10:26.266] [settingssync] [info] Using settings sync service https://vscode-sync-insiders.trafficmanager.net/
[2022-08-08 15:10:26.269] [settingssync] [info] Auto Sync is disabled.
```
And here's the output for `Log (Window)`:

```
[2022-08-08 15:10:26.609] [window] [info] Request to 'https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-server/0.1.2022080309/extension/package.nls.json' failed with status code 404
[2022-08-08 15:10:27.241] [window] [info] Request to 'https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-server/0.1.2022080309/extension/package.nls.json' failed with status code 404
[2022-08-08 15:10:27.243] [window] [info] Request to 'https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-server/0.1.2022080309/extension/package.nls.json' failed with status code 404
[2022-08-08 15:10:28.740] [window] [info] Request to 'https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-server/0.1.2022080309/extension/package.nls.json' failed with status code 404
[2022-08-08 15:10:28.795] [window] [info] Request to 'https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-server/0.1.2022080309/extension/package.nls.json' failed with status code 404
[2022-08-08 15:10:28.867] [window] [info] Request to 'https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-server/0.1.2022080309/extension/package.nls.json' failed with status code 404
[2022-08-08 15:10:52.979] [window] [info] Request to 'https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-server/0.1.2022080309/extension/package.nls.json' failed with status code 404
[2022-08-08 15:10:53.029] [window] [info] Request to 'https://ms-vscode.vscode-unpkg.net/ms-vscode/remote-server/0.1.2022080309/extension/package.nls.json' failed with status code 404
```

**Comment by bamurtaugh** [comment]
If Safari-only, I wonder if this is related to https://github.com/microsoft/vscode-remote-release/issues/7034?

**Comment by TylerLeonhardt** [comment]
@tlockney does your remote workspace resolve at all?

**Comment by tlockney** [comment]
@TylerLeonhardt Can you clarify what you mean by that? I can load the URL shown in the console when I start the server and when the UI loads in the browser, I can see the name of the remote server listed on the bottom left of the window. And I can even browse and open folders on the remote. But when I try to turn on settings sync, I get the 'no authentications provider' error mentioned above. At least I do in Safari, but it works fine in Chrome. 

For whatever it's worth, I've also verified this against both Apple Silicon machines and older Intel architecture, just in case. Same behavior in both cases.

**Comment by tlockney** [comment]
One additional data point that might be of interest, I just tried running in local mode (i.e., `code-server serve-local`) over an SSH tunnel to my remote machine and everything worked fine in this case, even in Safari.

**Comment by TylerLeonhardt** [comment]
I'm pretty suspicious of https://github.com/microsoft/vscode-remote-release/issues/7034 at this point.

**Comment by tlockney** [comment]
This seems to be working now based on whatever fix was applied to address #7034.
